### ISSUE TITLE
Actuarial: exposure curve upper bound hardcoded at 20x EV — truncates heavy-tailed distributions

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#822]
## Problem Statement
`LayerPricer.exposure_curve()` uses `max_loss = ev * 20.0` as the practical upper bound for the exposure curve (Lee diagram). For heavy-tailed distributions such as Pareto with alpha=2.5, the 99.9th percentile is approximately 31.6x the minimum value (xm), and for alpha=2.0, it is 31.6x xm, while the expected value is only 2*xm. This means the 99.9th percentile can be 15.8x EV, approaching the 20x cap.

For more extreme tail parameters (alpha < 2), the expected value may be infinite, causing the method to fall back to a diagonal line. Even when EV exists, the 20x cap systematically truncates the exposure curve for heavy-tailed distributions, understating the proportion of losses retained at high attachment points.

Reference: Lee (1988), Miccolis (1977), Klugman/Panjer/Willmot Chapter 5 on limited expected values.

## v1.0 Impact
Low-medium. The exposure curve is primarily a visualization tool, but inaccurate curves could mislead pricing decisions for high-attachment excess layers.

## Affected Code
- `ergodic_insurance/insurance_pricing.py:L306` — `max_loss = ev * 20.0` hardcoded upper bound

## Current Behavior
```python
max_loss = ev * 20.0
lev_max = self.severity.limited_expected_value(max_loss)
```

For Pareto(alpha=2.5, xm=100K): EV = $250K, max_loss = $5M
- True LEV at $5M covers ~98% of total expected losses
- But 99.9th percentile loss is ~$3.16M (barely within range)

For Pareto(alpha=1.5, xm=100K): EV = $300K, max_loss = $6M
- 99.9th percentile loss is ~$31.6M (5x above the cap!)
- Exposure curve is severely truncated

## Expected Behavior
The upper bound should be distribution-aware, using either:
1. A high quantile (e.g., 99.99th percentile) from the fitted distribution
2. A dynamically computed bound based on the distribution's survival function
3. At minimum, a configurable parameter instead of a hardcoded constant

## Recommended Approach
Add a `max_loss_quantile` parameter (default 0.9999) to `exposure_curve()` and compute `max_loss` from the distribution's inverse CDF when available, falling back to the current 20x heuristic.

## Acceptance Criteria
- [ ] `exposure_curve()` accepts configurable upper bound or quantile parameter
- [ ] Heavy-tailed distributions produce exposure curves extending to at least 99.9th percentile
- [ ] Fallback to 20x EV when inverse CDF is unavailable
- [ ] All existing tests continue to pass
- [ ] New test added to prevent regression
